mathics/core/expression.py

```diff
@@ -78,6 +78,48 @@
 )
 
 
+def build_elements_with_properties(self, elements: Iterable) -> tuple:
+    """Build a tuple of Elements noting useful properties such as
+    whether the collection of elements is sorted, flat, or fully
+    evaluated.
+
+    Note: we add or set the following fields:
+      self._elements_fully_evaluated, self._is_flat, and self._is_sorted
+
+    """
+
+    # All of the properties start out optimistic (True) and are reset when that proves wrong.
+
+    # _elements_fully_evaluated is True if all elements have been fully evaluated.
+    # Strings, and Numbers are fully evaluated. Symbols like Null, True, and False may be up for debate.
+    self._elements_fully_evaluated = True
+
+    # _is_flat is True if all elements are atoms/leaves.
+    self._is_flat = True
+
+    # _is_sorted is True if elements do not need sorting
+    # elements with less than 2 items or all have the same value are sorted.
+    self._is_sorted = True
+
+    result = []
+    last_element = None
+    for element in elements:
```

Comment:
Just one detail:  in master, Mathics calls ``from_python`` over each element. I do not know if this was a good idea, but we can get a change in the behavior if we do not do that here.